src/components/SettingsSection/BasicInformationSection.tsx

```diff
@@ -0,0 +1,252 @@
+import { useMemo, useState } from 'react';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { FormProvider, useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { basicInformationSchema } from '@/data/formSchemas';
+import { loadLanguages } from '@/data/languages';
+import { Box, Flex } from '@/styles/containers';
+import { Profile, ProfileLanguage } from '@/types/schema';
+import { formatEnumeration } from '@/utils/helpers';
+import { useProfileAuth } from '@/utils/hooks';
+import { SettingField, SettingSection } from '.';
+import CreatableBigDataDropdown from '../CreatableBigDataDropdown';
+import TextInput from '../TextInput';
+
+const getDefaults = (
+  profile: Partial<Profile>,
+  canSpeaks: ProfileLanguage[],
+  canReads: ProfileLanguage[],
+): Partial<z.infer<typeof basicInformationSchema>> => ({
+  firstName: profile.first_name,
+  lastName: profile.last_name,
+  country: profile.country,
+  state: profile.state,
+  city: profile.city,
+  phoneNumber: profile.phone_number,
+  canReads: canReads.map(lang => ({
+    label: lang.language_name,
+    value: lang.language_name,
+  })),
+  canSpeaks: canSpeaks.map(lang => ({
+    label: lang.language_name,
+    value: lang.language_name,
+  })),
+});
+
+export default function BasicInformationSection() {
+  const { profile, auth } = useProfileAuth();
+
+  const [isEditing, setIsEditing] = useState(false);
+
+  // split languages for rendering
+  const { languages: profileLanguages } = profile;
+  const canSpeakLangs = useMemo(
+    () => profileLanguages.filter(lang => lang.can_speak),
+    [profileLanguages],
+  );
+  const canReadLangs = useMemo(
+    () => profileLanguages.filter(lang => lang.can_read),
+    [profileLanguages],
+  );
+
+  const form = useForm<z.infer<typeof basicInformationSchema>>({
+    resolver: zodResolver(basicInformationSchema),
+    defaultValues: getDefaults(
+      profile.profileData ?? {},
+      canSpeakLangs,
+      canReadLangs,
+    ),
+  });
+
+  // handle save changes
+  const onValidSubmit = async (
+    values: z.infer<typeof basicInformationSchema>,
+  ) => {
+    const { userId } = auth;
+    if (!userId) throw new Error('Cannot update profile! Are you logged in?');
+
+    const allLangs: Record<string, { read: boolean; speak: boolean }> = {};
+    values.canReads.forEach(lang => {
+      try {
+        allLangs[lang.label].read = true;
+      } catch {
+        allLangs[lang.label] = { read: true, speak: false };
+      }
+    });
+    values.canSpeaks.forEach(lang => {
+      try {
+        allLangs[lang.label].speak = true;
```

Comment:
can i ask when/why this assignment would error?

Reply:
ye this can error if `allLangs[lang.label]` is undefined (i.e. the key references nothing), so attempting to do `undefined.speak` would error